rts/System/Platform/Linux/Futex.cpp

```diff
@@ -29,7 +33,11 @@ void spring_futex::lock()
 
 	do {
 		if ((c == 2) || __sync_val_compare_and_swap(&mtx, 1, 2) != 0)
+#ifndef __OpenBSD__
 			syscall(SYS_futex, &mtx, FUTEX_WAIT_PRIVATE, 2, NULL, NULL, 0);
+#else
+			futex(&mtx, FUTEX_WAIT_PRIVATE, 2, NULL, NULL);
+#endif
```

Comment:
Perhaps this pattern could be extracted to a function? So somewhere above there would be
```c
static inline void do_futex (uint32_t *mtx, uint32_t value)
{
	#ifndef __OpenBSD__
		syscall(SYS_futex, mtx, FUTEX_WAIT_PRIVATE, value, NULL, NULL, 0);
	#else
		futex(mtx, FUTEX_WAIT_PRIVATE, value, NULL, NULL);
	#endif
}
```
And this would become just
```suggestion
			do_futex(&mtx, 2);
```